run--01a--prep-geo-data.Rmd

```diff
@@ -260,168 +280,279 @@ schoolShp <-
+  download.file(url      = glue("https://nces.ed.gov/programs/edge/data/GRF23.zip"),
+            destfile = glue("{input_path}GRF23.zip"))
+  unzip(glue("{input_path}GRF23.zip"),
+        files = "grf23_lea_tract.xlsx",
         exdir = str_replace(input_path, "\\/$", ""))
 }
+
 geo_crosswalk_sd <-
   readxl::read_xlsx(grffile) %>% 
-  rename(school = NAME_LEA22,
+  rename(school = NAME_LEA23,
          GEOID  = TRACT) %>% 
   mutate(STATEFIP = substr(GEOID, 1, 2),
          LANDAREA = as.numeric(LANDAREA)) %>% 
-  filter(STATEFIP == {my_state_fip}) %>%
+  filter(STATEFIP == {my_state_fip})
+
+geo_crosswalk_sd <- geo_crosswalk_sd %>% 
+  merge(schoolShp %>% as.data.frame() %>% select(GEOID, level) %>% unique(),
+        by.x = "LEAID",
+        by.y = "GEOID",
+        all.x = TRUE) %>% 
+  filter(level %in% c("elementary", "unified")) %>% 
+  mutate(LAND_SQMETER  = LANDAREA*2589988.110336)
+
+geo_crosswalk_sd <- geo_crosswalk_sd %>%
   group_by(GEOID) %>% 
   mutate(pct_sd = LANDAREA / sum(LANDAREA))
+
 rm(grffile)
 ```
 
+```{r manually update school districts with the same name}
+#School districts with the same name
+#https://www.census.gov/programs-surveys/saipe/guidance-geographies/same-name/2023.html
+#/!\ Users should check whether their own state has this issue and manually update as needed following the example of Illinois below.
+if (my_state_abbr == "IL") {
+  schoolShp <- schoolShp %>% 
+    mutate(school = case_when(GEOID == "1729100" ~ "Oak Grove School District 68 (Peoria County)",
+                              GEOID == "1729130" ~ "Oak Grove School District 68 (Lake County)",
+                              TRUE               ~ school))
+  geo_crosswalk_sd <- geo_crosswalk_sd %>% 
+    mutate(school = case_when(LEAID == "1729100" ~ "Oak Grove School District 68 (Peoria County)",
+                              LEAID == "1729130" ~ "Oak Grove School District 68 (Lake County)",
+                              TRUE               ~ school))
+}
+```
+
+```{r check land area data}
+#compare land area from crosswalk to schoolShp
+geo_crosswalk_sd_area <- geo_crosswalk_sd %>% 
+  group_by(LEAID) %>% 
+  mutate(ALAND = sum(LAND_SQMETER))
+
+schoolShp_area <- schoolShp %>% 
+  select(GEOID, STATEFP, ELSDLEA, UNSDLEA, school, level, geometry, ALAND) %>% 
+  merge(geo_crosswalk_sd_area %>% as.data.frame() %>% select(LEAID, ALAND) %>% unique(),
+        by.x = "GEOID",
+        by.y = "LEAID",
+        all.x = TRUE) %>% 
+  mutate(pct_diff = (ALAND.x - ALAND.y)/ALAND.x)
+
+summary(schoolShp_area$pct_diff)
+
+if (FALSE) {
+  schoolShp_area %>% filter(is.na(pct_diff)) %>% View()
+}
+
+#check sum of land area to Census tract
+geo_crosswalk_sum_area <- geo_crosswalk_sd %>% 
+  group_by(GEOID) %>% 
+  mutate(sum_sd = sum(LAND_SQMETER)) %>% 
+  merge(tractShp %>% as.data.frame() %>% select(GEOID, ALAND) %>% unique(),
+        by = "GEOID",
+        all.x = TRUE) %>%
+  mutate(pct_diff = abs((ALAND - sum_sd)/sum_sd)) 
+
```

Comment:
Ditto above comment on more actively bringing user attention to sufficiently large discrepancies.